src/components/Courses/Courses.tsx

```diff
@@ -1,25 +1,39 @@
+import React from 'react';
 import { useState, useEffect } from 'react';
-import * as ImageService from '../services/api';
 import { useLocation } from 'react-router-dom';
+import * as ImageService from '../services/api';
 import Pagination from 'components/Pagination/Pagination';
 import { CoursesItem } from './CoursesItem/CoursesItem';
+import { Loader } from 'components/Loading/Loading';
 import {
   CoursesList,
   CourseItemWrapper,
   StyledLink,
   Title,
   CourseListWrapper,
 } from './Courses.module';
-import { Loader } from 'components/Loading/Loading';
+
+interface Course {
+  id: number;
+  title: string;
+  lessonsCount: number;
+  rating: number;
+  description: string;
+  meta: {
+    skills: string[];
+    courseVideoPreview: string;
+  };
+  previewImageLink: string;
+}
 
 export const Courses = () => {
-  const [courses, setCourses] = useState([]);
-  const [loading, setLoading] = useState(false);
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
```

Comment:
```suggestion
  const [loading, setLoading] = useState(false);
```
unnecessary overtyping